wandb/sdk/launch/runner/vertex_runner.py

```diff
@@ -180,11 +182,15 @@ def run(
         except Exception as e:
             raise LaunchError(f"Failed to create Vertex job: {e}")
         if synchronous:
-            job.run(**execution_kwargs, sync=True)
+            run = event_loop_thread_exec(job.run)
+            await run(**execution_kwargs, sync=True)
         else:
-            job.submit(**execution_kwargs)
+            submit = event_loop_thread_exec(job.submit)
+            await submit(**execution_kwargs)
         submitted_run = VertexSubmittedRun(job)
+        interval = 1
         while not getattr(job._gca_resource, "name", None):
             # give time for the gcp job object to be created and named, this should only loop a couple times max
-            time.sleep(1)
+            await asyncio.sleep(interval)
+            interval *= 2
```

Comment:
We should set a max interval.  In the edge case where we lose connectivity for just a couple minutes this could grow to a very long interval and would appear as if the agent is hung.  IMO `interval = min(30, interval * 2)`